### DISCUSSION TITLE
[v3] component directive not found in cra 5 project

### DISCUSSION THREAD

**Discussion by Lueton** [tailwindlabs/tailwindcss#6703] [Help]
**Dependencies**

```
    "autoprefixer": "^10.4.0",
    "postcss": "^8.4.5",
    "tailwindcss": "^3.0.7",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-scripts": "5.0.0",
```

**What version of Node.js are you using?**

`Node v16.13.1`

**What operating system are you using?**

`Windows`

**Reproduction URL**

[Reproduction Repository](https://github.com/Lueton/tailwind-3-directives-issue)

**Describe your issue**

Building my project always gives me this error:

`Syntax error: tailwind-3-directives-issue\src\components\button\button.css @layer components is used but no matching @tailwind components directive is present.`  

I tried using directives and imports, no difference. Did i miss something?

**Comment by Lueton** [comment]
Any suggestion? Is that just no longer supported?

**Comment by NovusOpus** [comment]


**Reply by Lueton** [reply]
Before v3 that wasn't a problem as long as you import the `button.css` into the `index.css` . That would mean that common folder structures like in my example would no longer work. Packing all of the CSS from each component into one file doesn't seem smart to me in large apps.

**Reply by NovusOpus** [reply]
Sorry my bad, I though you where testing Tailwind CSS and you've got a bit lost. 

But then I saw this was converted from an issue to a discussion, and it seems what you're trying to do is [Build-time imports](https://tailwindcss.com/docs/using-with-preprocessors#build-time-imports).

**Reply by Lueton** [reply]
Ah thank you, I didn't notice that I had to remove the `@layer components` directives in the ` button.css` and now only the import order is decisive.
